BUILD.bazel

```diff
@@ -1 +1,29 @@
-# This file exists so the root of our repository, e.g. the root `Cargo.toml`, can be referenced by Bazel.
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This file exists so the root of our repository, e.g. the root `Cargo.toml`, can be referenced by
+# Bazel.
+
+# Generates a file `git_version.rs` that includes a `pub const` with the current git hash.
+genrule(
+    name = 'git_version',
+    srcs = glob([
+        ".git/HEAD",
+        ".git/packed-refs",
+        ".git/refs/**",
+    ]),
+    outs = ['git_version.rs'],
+    cmd = """
+    git_hash=$$(git rev-parse --verify HEAD || echo "failed to get current git hash via Bazel genrule 'git_version'")
+    echo "pub const GIT_VERSION: &str = \\"$$git_hash\\";" > $(execpath git_version.rs)
+    """,
+    # TODO(parkmycar): I don't love using `local`, but otherwise this doesn't work.
```

Comment:
It seems like Bazel has a `workspace_status_command` feature that was built for extracting Git metadata. Did you look into that? Here's an example of using it (https://github.com/JulianSchmid/example-bazel-add-git-hash) that looks like it'd work for us.

Reply:
I did, at the moment this PR just seems a bit easier and achieves very similar behavior, but let me try wiring up the same thing with `workspace_status_command` and see how it goes!

Reply:
Updated this PR to use `workspace_status_command` and it feels a lot better! I'm glad we made this change